src/base/qtcompat.h

```diff
@@ -0,0 +1,49 @@
+/*
+ * Bittorrent Client using Qt and libtorrent.
+ * Copyright (C) 2017  Vladimir Golovnev <glassez@yandex.ru>
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * In addition, as a special exception, the copyright holders give permission to
+ * link this program with the OpenSSL project's "OpenSSL" library (or with
+ * modified versions of it that use the same license as the "OpenSSL" library),
+ * and distribute the linked executables. You must obey the GNU General Public
+ * License in all respects for all of the code used other than "OpenSSL".  If you
+ * modify file(s), you may extend this exception to your version of the file(s),
+ * but you are not obligated to do so. If you do not wish to do so, delete this
+ * exception statement from your version.
+ */
+
+#pragma once
+
+#include <QtGlobal>
+
+#if (QT_VERSION < QT_VERSION_CHECK(5,7,0))
+namespace QtPrivate
+{
+    template <typename T>
+    struct QAddConst
+    {
+        typedef const T Type;
+    };
+}
+
+// this adds const to non-const objects (like std::as_const)
+template <typename T>
+Q_DECL_CONSTEXPR typename QtPrivate::QAddConst<T>::Type &qAsConst(T &t) Q_DECL_NOTHROW { return t; }
```

Comment:
@glassez 
do you mind pointing out where this is being used?

Reply:
Oops... I first used `qAsConst` in the code, then I found out that it is only since Qt 5.8 and I stopped doing it. But then I decided to have this compatibility layer, since we will have all the same sooner or later to abandon `foreach`. But I forgot again to use it in the code! Fixed now.

Reply:
Damn. The discussed place caused an error, because I tried to change an iterable container. So this code is currently not used and I removed this file from this PR.